reverse_engineering/helpers/sequenceHelper.js

```diff
@@ -0,0 +1,290 @@
+/**
+ * @typedef {Function} Execute
+ * @typedef {{ [optionName: string]: string | number | undefined }} Options
+ */
+
+const {
+  Sequence,
+  SequenceDto,
+} = require('../../forward_engineering/types/schemaSequenceTypes');
+
+const START_VALUE_REGEX = /(?<=START WITH)(\s+)([\w]+)(?=\s+)?/;
+const SHARING_LINK_REGEX = /\s+LINK$/;
+
+/**
+ * @enum {string}
+ */
+const OPTION_VALUE = {
+  true: 'Y',
+  false: 'N',
+};
+
+/**
+ * @enum {string}
+ */
+const SEQUENCE_OPTION = {
+  cache: 'cache',
+  cycle: 'cycle',
+  extend: 'extend',
+  global: 'global',
+  keep: 'keep',
+  noCache: 'nocache',
+  noCycle: 'nocycle',
+  noExtend: 'noextend',
+  noKeep: 'nokeep',
+  noOrder: 'noorder',
+  noScale: 'noscale',
+  noShard: 'noshard',
+  order: 'order',
+  scale: 'scale',
+  scaleExtend: 'scaleExtend',
+  session: 'session',
+  shard: 'shard',
+  sharing: 'sharing',
+  type: 'type',
+};
+
+const SEQUENCE_OPTION_MAP = {
+  [SEQUENCE_OPTION.cycle]: {
+    [OPTION_VALUE.true]: SEQUENCE_OPTION.cycle,
+    [OPTION_VALUE.false]: SEQUENCE_OPTION.noCycle,
+  },
+  [SEQUENCE_OPTION.order]: {
+    [OPTION_VALUE.true]: SEQUENCE_OPTION.order,
+    [OPTION_VALUE.false]: SEQUENCE_OPTION.noOrder,
+  },
+  [SEQUENCE_OPTION.shard]: {
+    [OPTION_VALUE.true]: SEQUENCE_OPTION.shard,
+    [OPTION_VALUE.false]: SEQUENCE_OPTION.noShard,
+  },
+  [SEQUENCE_OPTION.scale]: {
+    [OPTION_VALUE.true]: SEQUENCE_OPTION.scale,
+    [OPTION_VALUE.false]: SEQUENCE_OPTION.noScale,
+  },
+  [SEQUENCE_OPTION.type]: {
+    [OPTION_VALUE.true]: SEQUENCE_OPTION.session,
+    [OPTION_VALUE.false]: SEQUENCE_OPTION.global,
+  },
+  [SEQUENCE_OPTION.keep]: {
+    [OPTION_VALUE.true]: SEQUENCE_OPTION.keep,
+    [OPTION_VALUE.false]: SEQUENCE_OPTION.noKeep,
+  },
+  [SEQUENCE_OPTION.extend]: {
+    [OPTION_VALUE.true]: SEQUENCE_OPTION.extend,
+    [OPTION_VALUE.false]: SEQUENCE_OPTION.noExtend,
+  },
+};
+
+/**
+ * @param {{ schema: string, execute: Execute, logger: object }}
+ * @returns {Promise<SequenceDto[]>}
+ */
+const getSchemaSequenceDtos = async ({ schema, execute, logger }) => {
+  try {
+    const queryResult = await execute(
```

Comment:
can we name more explicit? As we expect the result of this query to contain only `sequences`, let's use that.